### ISSUE TITLE
newui: DomPurify breaks self-closing tags

### ISSUE THREAD

**Issue by chloeleichen** [TerriaJS/terriajs#1411]
on newui, when using custom feature info template, we get warning such as: 
`Warning: img is a void element tag and must not have 'children' or use 'props.dangerouslySetInnerHTML'`. 

This is because dompurify changes the `<img/>` into `<img>`

Maybe we can list self closing tags as allowed tags in the setting? 

@RacingTadpole Can you have a look at this one please?

**Comment by RacingTadpole** [comment]
Sure thing.

**Comment by RacingTadpole** [comment]
This is an interesting one. DOMPurify ultimately adds elements using the browser's own functions, and then [uses `innerHTML` to get the html](https://github.com/cure53/DOMPurify/blob/master/src/purify.js#L752); as a result, it changes `<br />` to `<br>`. 
But React [requires all tags to close](https://facebook.github.io/react/tips/self-closing-tag.html) afaik.
No obvious solution is coming to me.

**Comment by RacingTadpole** [comment]
Fixed with a regex replace before calling `htmlToReactParser`: https://github.com/TerriaJS/terriajs/commit/09be3b5eee62955b18482699034ced68afa2c69a

**Comment by RacingTadpole** [comment]
Hmm - actually that hasn't solved it. Even with the above fix, somehow `<br/>` winds up with 2 `props.children`, albeit empty children: 
![image](https://cloud.githubusercontent.com/assets/1757858/13942882/69f90100-f04b-11e5-9992-7d30b3d1bb2e.png)

**Comment by RacingTadpole** [comment]
Fixed - the PR to fix this problem already existed in `html-to-react`, so we now use a fork where it is merged. Once it is merged in master, we can go back to the standard package.

**Comment by duhrmacher** [comment]
We are experiencing this issue now.

**Comment by zoran995** [comment]
Hi @duhrmacher I think it would be best to open a new issue with more details on what are you seeing, including version of terriajs/terriamap you are using. This issue is over 8 years old and lot of things have changed since then in terriajs codebase
